### DISCUSSION TITLE
Park UI v.s. Chakra UI

### DISCUSSION THREAD

**Discussion by zachguo** [cschroeter/park-ui#221] [General]
Is the bundle size of Park UI much smaller than Chakra UI?
Are there any pitfalls I should be aware of if I want to migrate from Chakra UI to Park UI?

**Comment by cschroeter** [comment]
@zachguo 

Thanks for taking your time to start this discussion. Park UI is not a classical UI library like Chakra or Material UI. I've tried my best to summarize this in the [introduction](https://park-ui.com/docs/panda/overview/introduction) for new users.

Under the hood Park UI uses the headless component library [Ark UI](https://ark-ui.com/) and uses various different styling engines like Tailwind or Panda  CSS to style them.  So if you plan to replace your existing app the best approach would be React + Panda CSS.

However we also plan to bring the Ark headless components to Chakra UI in the next weeks.

**Reply by zachguo** [reply]
Yup, I started using Ark UI to replace a few heavy components from Chakra.

Migration of theming and style props from Chakra to Park UI(or Panda CSS) looks like a daunting task so I gave up.

My current plan is 
- replace all Chakra components with Ark components with my own vanilla CSS utilizing Chakra's internal CSS variables.
- once all Chakra components are replaced, I'll define those CSS variables directly in global CSS to replace Chakra's theme provider.

It would be nicer if Ark UI could show each component's data attributes in the docs, and provide a snippet of vanilla CSS to produce the demo component shown in the docs. Right now I have to dig into the source to figure out how to style each Ark component.
